clumper/clump.py

```diff
@@ -950,6 +954,92 @@ def head(self, n=5):
         n = min(n, len(self))
         return self._create_new([self.blob[i] for i in range(n)])
 
+    def sample(
+        self,
+        n: int,
+        replace: bool,
+        random_state: int,
+        weights: List[str] = None,
+    ):
+        """Samples n data from the collection
+
+        Args:
+            n (int): The number of items to sample
+            replace (bool): Have duplicate items or not. Defaults to False.
+            weights (List[str], optional): The key(s) used for calculate sample probability. Defaults to None which means equal probability
```

Comment:
How could a list of str make sense here? We can only assign one column as a sampling weight no?

Reply:
That's a good point. I complicated it by thinking that we wanted to assign multiple columns as keys.